homeassistant/components/denonavr/media_player.py

```diff
@@ -73,99 +54,30 @@
-        vol.Optional(CONF_NAME): cv.string,
-    }
-)
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_HOST): cv.string,
-        vol.Optional(CONF_NAME): cv.string,
-        vol.Optional(CONF_SHOW_ALL_SOURCES, default=DEFAULT_SHOW_SOURCES): cv.boolean,
-        vol.Optional(CONF_ZONES): vol.All(cv.ensure_list, [DENON_ZONE_SCHEMA]),
-        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
-    }
-)
 
-NewHost = namedtuple("NewHost", ["host", "name"])
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Denon platform."""
-    # Initialize list with receivers to be started
-    receivers = []
-
-    cache = hass.data.get(KEY_DENON_CACHE)
-    if cache is None:
-        cache = hass.data[KEY_DENON_CACHE] = set()
-
-    # Get config option for show_all_sources and timeout
-    show_all_sources = config[CONF_SHOW_ALL_SOURCES]
-    timeout = config[CONF_TIMEOUT]
-
-    # Get config option for additional zones
-    zones = config.get(CONF_ZONES)
-    if zones is not None:
-        add_zones = {}
-        for entry in zones:
-            add_zones[entry[CONF_ZONE]] = entry.get(CONF_NAME)
-    else:
-        add_zones = None
-
-    # Start assignment of host and name
-    new_hosts = []
-    # 1. option: manual setting
-    if config.get(CONF_HOST) is not None:
-        host = config.get(CONF_HOST)
-        name = config.get(CONF_NAME)
-        new_hosts.append(NewHost(host=host, name=name))
-
-    # 2. option: discovery using netdisco
-    if discovery_info is not None:
-        host = discovery_info.get("host")
-        name = discovery_info.get("name")
-        new_hosts.append(NewHost(host=host, name=name))
-
-    # 3. option: discovery using denonavr library
-    if config.get(CONF_HOST) is None and discovery_info is None:
-        d_receivers = denonavr.discover()
-        # More than one receiver could be discovered by that method
-        for d_receiver in d_receivers:
-            host = d_receiver["host"]
-            name = d_receiver["friendlyName"]
-            new_hosts.append(NewHost(host=host, name=name))
-
-    for entry in new_hosts:
-        # Check if host not in cache, append it and save for later
-        # starting
-        if entry.host not in cache:
-            new_device = denonavr.DenonAVR(
-                host=entry.host,
-                name=entry.name,
-                show_all_inputs=show_all_sources,
-                timeout=timeout,
-                add_zones=add_zones,
-            )
-            for new_zone in new_device.zones.values():
-                receivers.append(DenonDevice(new_zone))
-            cache.add(host)
-            _LOGGER.info("Denon receiver at host %s initialized", host)
-
-    # Add all freshly discovered receivers
-    if receivers:
-        add_entities(receivers)
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Xiaomi Gateway Alarm from a config entry."""
```

Comment:
`Xiaomi Gateway Alarm`?

Reply:
sorry that was old code, I based this component on my previous PR of the Xiaomi Gateway Alarm.
just pushed a Commit to fix this

Reply:
You don't have to be sorry. Copy/paste rulez ;)